src/instructlab/cli/model/chat.py

```diff
@@ -0,0 +1,269 @@
+)
+@click.option(
+    "--tls-client-key",
+    type=click.Path(),
+    default="",
+    show_default=True,
+    help="Path to the TLS client key to use.",
+)
+@click.option(
+    "--tls-client-passwd",
+    type=click.STRING,
+    default="",
+    help="TLS client certificate password.",
+)
+@click.option(
+    "--model-family",
+    help="Force model family to use when picking a chat template",
+)
+@click.option(
+    "--serving-log-file",
+    type=click.Path(path_type=pathlib.Path),
+    required=False,
+    help="Log file path to write server logs to.",
+)
+@click.option(
+    "-t",
+    "--temperature",
+    cls=clickext.ConfigOption,
+)
+@click.pass_context
+@clickext.display_params
+def chat(
+    ctx,
+    question,
+    model,
+    context,
+    session,
+    quick_question,
+    max_tokens,
+    endpoint_url,
+    api_key,
+    tls_insecure,
+    tls_client_cert,
+    tls_client_key,
+    tls_client_passwd,
+    model_family,
+    serving_log_file,
+    temperature,
+):
+    """Runs a chat using the modified model"""
+    # pylint: disable=import-outside-toplevel
+    # First Party
+    from instructlab.model.backends.common import is_temp_server_running
+    from instructlab.model.chat import ChatException, chat_cli
+
+    users_endpoint_url = cfg.get_api_base(ctx.obj.config.serve.host_port)
+
+    # we prefer the given endpoint when one is provided, else we check if the user
+    # is actively serving something before falling back to serving our own model
+    backend_instance = None
+    if endpoint_url:
+        api_base = endpoint_url
+    elif is_openai_server_and_serving_model(
+        users_endpoint_url,
+        api_key,
+        http_params={
+            "tls_client_cert": tls_client_cert,
+            "tls_client_key": tls_client_key,
+            "tls_client_passwd": tls_client_passwd,
+            "tls_insecure": tls_insecure,
+        },
+    ):
+        api_base = users_endpoint_url
+        if serving_log_file:
+            logger.warning(
+                "Setting serving log file (--serving-log-file) is not supported when the server is already running"
+            )
+    else:
+        # First Party
+        from instructlab.model.backends import backends
+
+        # If a log file is specified, write logs to the file
+        root_logger = logging.getLogger()
+        if serving_log_file:
+            log.add_file_handler_to_logger(root_logger, serving_log_file)
+
+        ctx.obj.config.serve.llama_cpp.llm_family = model_family
+        backend_instance = backends.select_backend(
+            ctx.obj.config.serve,
+            model_path=model,
+            log_file=serving_log_file,
+        )
+        try:
+            # Run the llama server
+            api_base = backend_instance.run_detached(http_client(ctx.params))
+        except Exception as exc:
+            click.secho(f"Failed to start server: {exc}", fg="red")
+            raise click.exceptions.Exit(1)
```

Comment:
server invocation is also something i think maybe better left in the backend so that other clients can also trigger this action